packages/timeline-state-resolver/src/conductor.ts

```diff
@@ -607,6 +621,18 @@ export class Conductor extends EventEmitter<ConductorEvents> {
 				return Promise.reject(`No matching device type for "${type}" ("${DeviceType[type]}") found in conductor`)
 			}
 
+			newDevice = await makeAbortable(async () => {
+				throwIfAborted()
+				const newDevice = await newDevicePs
+				if (options?.signal?.aborted) {
```

Comment:
Is this if-statement necessary? 
We check if we are aborted two lines above this and since `options.signal` is a parameter of the method we are in, then I currently don't see how `newDevicePs` should be able to alter it.

Reply:
`newDevicePs` is not abortable in itself. If it runs longer that expected, and the AbortController.abort() gets called somewhere in the middle. The promise returned by `makeAbortable` will throw immediately and let code that awaits `createDevice` proceed, but `newDevicePs` unfortunatelly will keep running unless it settles, and only then we can do the eventual cleanup.